src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -138,9 +138,8 @@ internal static int OptimalMaxSpinWaitsPerSpinIteration
                 return 7;
             }
         }
-
 #if TARGET_UNIX || TARGET_BROWSER
-        internal WaitSubsystem.ThreadWaitInfo WaitInfo => _waitInfo;
+        internal WaitSubsystem.ThreadWaitInfo WaitInfo => EnsureWaitInfo();
```

Comment:
```suggestion
        internal WaitSubsystem.ThreadWaitInfo WaitInfo => return Volatile.Read(ref _waitInfo) ?? AllocateWaitInfo();
```

Reply:
It was called from another place in code, so I kept it since it was inlined, but sure we can drop it and access property instread in InitializeCurrentThread.

Reply:
Does that another place still need to call `EnsureWaitInfo` now that it is always allocated lazily?

Reply:
No, I dropped it in last commit.